src/Marten/Events/Projections/ViewProjection.cs

```diff
@@ -680,12 +680,12 @@ public enum ProjectionEventType
         {
             if (viewIdSelector == null && viewIdsSelector == null)
                 throw new ArgumentException($"{nameof(viewIdSelector)} or {nameof(viewIdsSelector)} must be provided.");
-            if (handler == null && type == ProjectionEventType.CreateAndUpdate)
```

Comment:
@SimonCropp existing code is actually throwing an `ArgumentNullException` for the case of `handler` being `null` and type being `ProjectionEventType.CreateAndUpdate`. But the code changes looks to throw an exception for `handler` being `null` and for `type` being `ProjectionEventType.CreateAndUpdate` or `type == ProjectionEventType.UpdateOnly`. This changes the existing behaviour event though unit tests are green. Thoughts?